pkg/kube/snapshot/snapshot_alpha.go

```diff
@@ -0,0 +1,317 @@
+				"snapshotClassName":   source.VolumeSnapshotClassName,
+			},
+		},
+	}
+
+	_, err = sna.dynCli.Resource(VolSnapContentGVR).Namespace("").Create(content, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Errorf("Failed to create content, VolumesnapshotContent: %s, Error: %v", content.GetName(), err)
+	}
+	_, err = sna.dynCli.Resource(VolSnapGVR).Namespace(namespace).Create(snap, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Errorf("Failed to create content, Volumesnapshot: %s, Error: %v", snap.GetName(), err)
+	}
+	if !waitForReady {
+		return nil
+	}
+
+	err = sna.WaitOnReadyToUse(ctx, snapshotName, namespace)
+	return err
+}
+
+// WaitOnReadyToUse will block until the Volumesnapshot in namespace 'namespace' with name 'snapshotName'
+// has status 'ReadyToUse' or 'ctx.Done()' is signalled.
+func (sna *SnapshotAlpha) WaitOnReadyToUse(ctx context.Context, snapshotName, namespace string) error {
+	return poll.Wait(ctx, func(context.Context) (bool, error) {
+		us, err := sna.dynCli.Resource(VolSnapGVR).Namespace(namespace).Get(snapshotName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		vs := v1alpha1.VolumeSnapshot{}
+		err = TransformUnstructured(us, &vs)
+		if err != nil {
+			return false, err
+		}
+		// Error can be set while waiting for creation
+		if vs.Status.Error != nil {
+			return false, errors.New(vs.Status.Error.Message)
+		}
+		return (vs.Status.ReadyToUse && vs.Status.CreationTime != nil), nil
+	})
+}
+
+func (sna *SnapshotAlpha) getContent(ctx context.Context, contentName string) (*v1alpha1.VolumeSnapshotContent, error) {
+	us, err := sna.dynCli.Resource(VolSnapContentGVR).Namespace("").Get(contentName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	vsc := v1alpha1.VolumeSnapshotContent{}
+	err = TransformUnstructured(us, &vsc)
+	if err != nil {
+		return nil, err
+	}
+	return &vsc, nil
+}
+
+func (sna *SnapshotAlpha) getDeletionPolicyFromClass(snapClassName string) (string, error) {
+	us, err := sna.dynCli.Resource(VolSnapClassGVR).Namespace("").Get(snapClassName, metav1.GetOptions{})
+	if err != nil {
+		return "", errors.Wrapf(err, "Failed to find VolumeSnapshotClass: %s", snapClassName)
+	}
+	vsc := v1alpha1.VolumeSnapshotClass{}
+	err = TransformUnstructured(us, &vsc)
+	if err != nil {
+		return "", err
+	}
+	return vsc.DeletionPolicy, nil
+}
+
+func (sna *SnapshotAlpha) createVolumeSnapshot(name, namespace string, pvcObjectRef corev1.ObjectReference, snapClassName string) error {
+	snap := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": fmt.Sprintf("%s/%s", v1alpha1.GroupName, v1alpha1.Version),
+			"kind":       VolSnapKind,
+			"metadata": map[string]interface{}{
+				"name":      name,
+				"namespace": namespace,
+			},
+			"spec": map[string]interface{}{
+				"source": map[string]interface{}{
+					"kind":      pvcObjectRef.Kind,
+					"name":      pvcObjectRef.Name,
+					"namespace": pvcObjectRef.Namespace,
```

Comment:
in sna.Create you are not calling createVolumeSnapshot with the pvcObjectRef.Namespace field set. Either set it there or use the namespace passed in the arguments.

Reply:
you are still passing in a pvcObjectRef with namspace not set in sna.Create. See line 87. Let me know if I'm missing something before resolving

Reply:
Aah, sorry, let me fix it